src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -186,15 +213,31 @@ auto StreamReader::decode_range(size_t begin_idx, size_t end_idx) const -> Decod
                 results.as_handle(),
                 message.c_str(),
                 log_event.get_timestamp(),
-                log_level,
-                log_num
+                log_event.get_log_level(),
+                log_event_idx + 1
         );
-        ++log_num;
     }
 
     return DecodedResultsTsType(results);
 }
 
+void StreamReader::filter_log_events(emscripten::val const& log_level_filter) {
```

Comment:
This should be after `get_filtered_log_event_map` (order in the header file should match order in the implementation file).